Feature/scraper

Ciuffi opened the pull request:
Updated API to work faster and be more accessible

Ciuffi committed:
First working scraper commit

Ciuffi committed:
Allowed for responding with json if it isn't too big.

Ciuffi committed:
Added Procfile for heroku

Ciuffi committed:
Added Procfile for heroku

Ciuffi committed:
Added Procfile for heroku

Ciuffi committed:
Added Procfile for heroku

Ciuffi committed:
Added event blocker for Scrape.

Ciuffi committed:
Added Procfile for heroku
Added a UI.

Ciuffi committed:
Added Procfile for heroku
Added a UI.

Ciuffi committed:
Added Procfile for heroku
Added a UI.

Ciuffi committed:
Added Procfile for heroku
Added a UI and setup for heroku

Ciuffi committed:
Better async and function organization

Ciuffi committed:
Better async and function organization

Ciuffi committed:
Better async and function organization

Fixed type in Scraper.js and fixed Socket.Io for any domain

calvinywcheng reviewed (commented):
You might want to consider decoupling the scraping/updating of courses and the getting of the course list... as in, have an API endpoint to scrape all the courses and store them in a database

and a number of API endpoints to navigate through the course lists (which would pull data from your database, as opposed to re-scraping the pages)

calvinywcheng commented on html/index.html:
```diff
@@ -0,0 +1,32 @@
+<input id="size"/>
+<p>search: </p>
+<input id="search"/>
+<button onclick="scrape();">Click me!</button>
```
`onclick` tags are generally considered bad practice--you can read up on [Unobtrusive Javascript](http://blog.teamtreehouse.com/unobtrusive-javascript-important)

Essentially, you want to separate the layout of the page (which is done via HTML) from the behaviours/interactions (which is done via JS)

Ciuffi committed:
Added postgress DB support
Scraper now stores data into DB instead of into system memory
added postgres module
Cleaned up server.js
Added tables.txt for creating postgres tables

Ciuffi committed:
Merge remote-tracking branch 'origin/Feature/Scraper' into Feature/Scraper

# Conflicts:
#	package.json
#	scripts/Scraper.js
#	server.js

Ciuffi committed:
add database schema file

Ciuffi committed:
Fix UBC connection resets

Ciuffi committed:
update database schema file to TAR encoding

Ciuffi committed:
Fix Faculty mispelling in db schema

Ciuffi committed:
Update scraper to work with new schema

Ciuffi committed:
Merge branch 'Feature/Scraper' of https://github.com/Ciuffi/UBCBetterCourseSelection into Feature/Scraper

Ciuffi committed:
remove old schema file
